kmk/hid.py

```diff
@@ -254,26 +301,25 @@ def has_key(self, key):
 
 
 class USBHID(AbstractHID):
-    REPORT_BYTES = 9
+    report_bytes_default = 9
+    REPORT_BYTES = report_bytes_default
 
     def __init__(self, **kwargs):
+        self.hid = usb_hid
+        self.hid_devices = self.hid.devices
+        self.usb_status = None
+        super().__init__(**kwargs)
 
-        self.devices = {}
-
-        for device in usb_hid.devices:
-            us = device.usage
-            up = device.usage_page
-
-            if up == HIDUsagePage.CONSUMER and us == HIDUsage.CONSUMER:
-                self.devices[HIDReportTypes.CONSUMER] = device
-            elif up == HIDUsagePage.KEYBOARD and us == HIDUsage.KEYBOARD:
-                self.devices[HIDReportTypes.KEYBOARD] = device
-            elif up == HIDUsagePage.MOUSE and us == HIDUsage.MOUSE:
-                self.devices[HIDReportTypes.MOUSE] = device
-            elif up == HIDUsagePage.SYSCONTROL and us == HIDUsage.SYSCONTROL:
-                self.devices[HIDReportTypes.SYSCONTROL] = device
+    def watchdog(self):
+        if self.usb_status != supervisor.runtime.usb_connected:
+            self.usb_status = supervisor.runtime.usb_connected
+            self.find_devices()
+            self.setup_keyboard_hid()
+            self.setup_mouse_hid()
+            self.show_debug()
```

Comment:
```suggestion
    def _connected(self):
        return supervisor.runtime.usb_connected
        # return self.ble.connected # for BLE

    def watchdog(self):
        if self.connected != self._connected():
            self.connected = self._connected()
            self.find_devices()
            self.setup_keyboard_hid()
            self.setup_mouse_hid()
            self.show_debug()
```
Make a wrapper to check for connection, then use that wrapper in the watchdog and we can put the watchdog implementation into the base class.
This also triggers on disconnects / reconnects and reallocates buffers.

Reply:
This makes sense for the future, but as of right now that would be wasted on BLE since we don't support custom descriptors there.